packages/react-storage/src/components/StorageBrowser/types.ts

```diff
@@ -61,33 +61,35 @@ export interface StorageBrowserProviderProps<V = {}>
   views?: V;
 }
 
-export interface StorageBrowserType<C extends ExtendedActionConfigs> {
-  (
-    props: StorageBrowserProps<DerivedActionViewType<C>, DerivedCustomViews<C>>
-  ): React.JSX.Element;
+export interface StorageBrowserType<K = string, V = {}> {
+  (props: StorageBrowserProps<K, V>): React.JSX.Element;
   displayName: string;
-  Provider: (
-    props: StorageBrowserProviderProps<DerivedCustomViews<C>>
-  ) => React.JSX.Element;
+  Provider: (props: StorageBrowserProviderProps<V>) => React.JSX.Element;
   CopyView: CopyViewType;
   CreateFolderView: CreateFolderViewType;
   DeleteView: DeleteViewType;
   UploadView: UploadViewType;
-  LocationActionView: LocationActionViewType<DerivedActionViewType<C>>;
+  LocationActionView: LocationActionViewType<K>;
   LocationDetailView: LocationDetailViewType;
   LocationsView: LocationsViewType;
 }
 
 type DefaultActionType<T = string> = Exclude<T, keyof DefaultActionConfigs>;
 
-export type DerivedCustomViews<T extends StorageBrowserActions> = {
+/**
+ * @internal
+ **/
```

Comment:
Comment syntax s slightly off and let's go ahead and mark these as `@unstable` as well:
```suggestion
/**
 * @internal @unstable
 */
```